scheduling-manager/src/main/java/com/amazonaws/blox/scheduling/scheduler/engine/daemon/DaemonEnvironment.java

```diff
@@ -35,8 +36,12 @@
 
   private final EnvironmentDescription environment;
 
-  public boolean hasMatchingTask(List<Task> tasks) {
-    return tasks.stream().noneMatch(this::matchesTask);
+  public boolean needsToAssign(List<Task> tasks) {
+    return tasks.stream().noneMatch(this::taskDesiredToRun);
+  }
+
+  public boolean noRunningTasks(List<Task> tasks) {
+    return tasks.stream().noneMatch(t -> HEALTHY_STATES.contains(t.getStatus()));
```

Comment:
The method name says `Running`, but the implementation says `HEALTHY`, so maybe `noHealthyTasks`?

Reply:
Sure. (Yes I realize that there are two Healthy states...)